### ISSUE TITLE
Problem with keys containing numbers

### ISSUE THREAD

**Issue by kwellman** [icanzilb/JSONModel#364]
I have a key called score1 and score2 in my json. My payload gets converted properly to an NSDictionary in my request handler but when I try to create a JSONModel instance using the initWithDictionary, I get an "Keys missing" error for those keys. When I use something like firstScore and secondScore as key names it works fine.

My model interface contains the properties:

@property (assign, nonatomic) int score1;
@property (assign, nonatomic) int score2;

Is there a problem when using key names with numbers in them?

EDIT: After investigating further it seems the keys are being transformed to score_1 and score_2 by the [JSONKeyMapper mapperFromUnderscoreCaseToCamelCase]. Is this the intended result?

**Comment by dabvid** [comment]
I also experience this issue.

**Comment by sauvikatinnofied** [comment]
Will you please share the sample JSON coming from the server?

**Comment by dabvid** [comment]
Absolutely. Like the creator of this issue I also use [JSONKeyMapper mapperFromUnderScoreCaseToCamelCase]

"address_info" :  {
                    "address1": "Streetname 1",
                    "address2":"",
                    "city":"",
                    "country":"",
                    "geo" : "54.7141546,13.216793499999994",
                    region : "",
                    zipcode: 21199
                };

The mapper is trying to map address1 to address_1.

**Comment by sauvikatinnofied** [comment]
I don't know whether adding the following code will be helpful to you or or.

**Comment by dabvid** [comment]
Thanks. I have tried this in my global key mapper:

```
[JSONModel setGlobalKeyMapper:[[JSONKeyMapper alloc] initWithDictionary:@{@"address1":@"address1", @"address2":@"address2""}]];
```

My local model key mapper needs to use:

```
+ (JSONKeyMapper*)keyMapper {
    return [JSONKeyMapper mapperFromUnderscoreCaseToCamelCase];
}
```

But I will try out your suggestion and get back with reply. Thanks.

**Comment by billinghamj** [comment]
Indeed, it is because of the key mapper. This behavior is not "broken" as such - it's just the way that particular key mapper is designed to work.

PR #332 would probably be the best solution for your use-case. Once it is merged, you could use the same code here.

**Comment by dabvid** [comment]
@billinghamj Thanks for pointing to the PR. The mapperFromUnderscoreCaseToCamelCase prefixes the numbers in keys with a  "_". Declaring a property with the same prefix (eg. address_1) still does not map correctly. Would you say there there is some way to use the mapperFromUnderscoreCaseToCamelCase with keys containing numbers somehow without waiting for the PR to be merged?

**Comment by billinghamj** [comment]
If you'd like a very short term solution, you could copy the code in the PR and make a category on the key mapper.

**Comment by billinghamj** [comment]
Resolved: b417d3f40603d64870f160b500e699b711f2d43b
